tests/nlu/test_disjoint_dialogue_act_extractor.py

```diff
@@ -0,0 +1,58 @@
+"""Tests for disjoint dialogue act extractor."""
+
+from unittest.mock import Mock
+
+import pytest
+
+from dialoguekit.core.intent import Intent
+from dialoguekit.core.slot_value_annotation import SlotValueAnnotation
+from dialoguekit.core.utterance import Utterance
+from dialoguekit.nlu.disjoint_dialogue_act_extractor import (
+    DisjointDialogueActExtractor,
+)
+from dialoguekit.participant.participant import DialogueParticipant
+
+
+@pytest.fixture
+def intent_classifier() -> Mock:
+    """Mock intent classifier."""
+    intent_classifier = Mock()
+    intent_classifier.classify_intent.return_value = Intent("DISCLOSE")
+    return intent_classifier
+
+
+@pytest.fixture
+def slot_value_annotator() -> Mock:
+    """Mock slot value annotator."""
+    slot_value_annotator = Mock()
+    slot_value_annotator.get_annotations.return_value = [
+        SlotValueAnnotation("GENRE", "action")
+    ]
+    return slot_value_annotator
+
+
+@pytest.fixture
+def dialogue_act_extractor(
+    intent_classifier: Mock, slot_value_annotator: Mock
+) -> DisjointDialogueActExtractor:
+    """Dialogue act extractor fixture."""
+    return DisjointDialogueActExtractor(
+        intent_classifier, [slot_value_annotator]
+    )
+
+
+def test_extract_dialogue_acts(
```

Comment:
Is there always a dialogue act, as in, do we have an "Other" (or similar) fall-back intent in case the classifier cannot predict anything else?

Reply:
Good point. I updated `DisjointDialogueActExtractor` to return an empty list if no intent are identified. Note that `classify_intent` is abstract in `IntentClassifier` so it gives the opportunity to define a fall-back intent for specific use cases.